llpc/translator/lib/SPIRV/SPIRVReader.cpp

```diff
@@ -4686,6 +4686,18 @@ Value *SPIRVToLLVM::transValueWithoutDecoration(SPIRVValue *bv, Function *f, Bas
     auto br = static_cast<SPIRVBranchConditional *>(bv);
     auto c = transValue(br->getCondition(), f, bb);
 
+    // OpBranchConditional can branch with OpUndef as condition. The selected jump target is undefined.
+    // In LLVM IR, BranchInst with undef value is fully UB.
+    // LLVM will mark the successors as unreachable, so later SimplifyCFG passes will cause
+    // unwanted removal of branches. By using a freeze instruction as condition, we make sure that LLVM
+    // will operate on a fixed value.
+    if (isa<UndefValue>(c)) {
+      auto builder = getBuilder();
+      IRBuilder<>::InsertPointGuard guard(*builder);
```

Comment:
Is this guard needed? I would not expect `CreateFreeze` to change the insertion point.